## `nixpkgs-review` result

Generated using [`nixpkgs-review-gha`](https://github.com/Defelo/nixpkgs-review-gha)

Command: `nixpkgs-review pr 421536`

Logs: https://github.com/GaetanLepage/nixpkgs-review-gha/actions/runs/16011238446


---
### `x86_64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.schedula</li>
    <li>python312Packages.schedula.dist</li>
    <li>python313Packages.schedula</li>
    <li>python313Packages.schedula.dist</li>
  </ul>
</details>


---
### `aarch64-linux` (sandbox = true)
<details>
  <summary>:white_check_mark: 4 packages built:</summary>
  <ul>
    <li>python312Packages.schedula</li>
    <li>python312Packages.schedula.dist</li>
    <li>python313Packages.schedula</li>
    <li>python313Packages.schedula.dist</li>
  </ul>
</details>


---
### `x86_64-darwin` (sandbox = true)
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.schedula</li>
    <li>python312Packages.schedula.dist</li>
    <li>python313Packages.schedula</li>
    <li>python313Packages.schedula.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `x86_64-darwin`</summary>
<details>
<summary>python312Packages.schedula</summary>
<pre>    rv = reductor(4)

tests/test_dispatcher.py: 98 warnings
tests/test_readme.py: 6 warnings
  /nix/store/laznlnxwn0rzw6w1sw1ahcfj63fy74v9-python3.12-multiprocess-0.70.18/lib/python3.12/site-packages/multiprocess/popen_fork.py:66: DeprecationWarning: This process (pid=8954) is multi-threaded, use of fork() may lead to deadlocks in the child.
    self.pid = os.fork()

tests/utils/test_blue.py: 16 warnings
tests/utils/test_io.py: 4 warnings
  /nix/store/q2y1zwszjmsap7r6nbk4dxbcsm47abfx-python3-3.12.11/lib/python3.12/pickle.py:576: DeprecationWarning: Pickle, copy, and deepcopy support will be removed from itertools in Python 3.14.
    rv = reduce(self.proto)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_readme.py::TestDoctest::runTest - AssertionError: 6 != 0 : (6, 55)
FAILED tests/utils/test_base.py::TestDoctest::runTest - AssertionError: 4 != 0 : (4, 25)
FAILED tests/utils/test_drw.py::TestDispatcherDraw::test_view - KeyError: 'plot'
FAILED tests/utils/test_web.py::TestDispatcherWeb::test_web - PermissionError: [Errno 1] Operation not permitted
FAILED tests/utils/test_web.py::TestDispatcherWeb::test_web_debug - PermissionError: [Errno 1] Operation not permitted
===== 5 failed, 95 passed, 1 deselected, 234 warnings in 70.25s (0:01:10) ======</pre>
</details>
<details>
<summary>python313Packages.schedula</summary>
<pre>
tests/test_dispatcher.py: 98 warnings
tests/test_readme.py: 6 warnings
  /nix/store/cg7g1gyq7cnsq0skfxy7mn95naag62ra-python3.13-multiprocess-0.70.18/lib/python3.13/site-packages/multiprocess/popen_fork.py:67: DeprecationWarning: This process (pid=9077) is multi-threaded, use of fork() may lead to deadlocks in the child.
    self.pid = os.fork()

tests/utils/test_blue.py: 16 warnings
tests/utils/test_io.py: 4 warnings
  /nix/store/il3s6rjh49l9ccg1nlqm93w67lcnfmqn-python3-3.13.4/lib/python3.13/pickle.py:576: DeprecationWarning: Pickle, copy, and deepcopy support will be removed from itertools in Python 3.14.
    rv = reduce(self.proto)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_dispatcher.py::TestAsyncParallel::test_multiple - AssertionError: 5.168652057647705 not less than 5.089165210723877
FAILED tests/test_readme.py::TestDoctest::runTest - AssertionError: 6 != 0 : (6, 58)
FAILED tests/utils/test_base.py::TestDoctest::runTest - AssertionError: 4 != 0 : (4, 25)
FAILED tests/utils/test_drw.py::TestDispatcherDraw::test_view - KeyError: 'plot'
FAILED tests/utils/test_web.py::TestDispatcherWeb::test_web - PermissionError: [Errno 1] Operation not permitted
FAILED tests/utils/test_web.py::TestDispatcherWeb::test_web_debug - PermissionError: [Errno 1] Operation not permitted
===== 6 failed, 94 passed, 1 deselected, 234 warnings in 69.92s (0:01:09) ======</pre>
</details>
</details>


---
### `aarch64-darwin` (sandbox = true)
<details>
  <summary>:x: 4 packages failed to build:</summary>
  <ul>
    <li>python312Packages.schedula</li>
    <li>python312Packages.schedula.dist</li>
    <li>python313Packages.schedula</li>
    <li>python313Packages.schedula.dist</li>
  </ul>
</details>

---
<details>
<summary>Error logs: `aarch64-darwin`</summary>
<details>
<summary>python312Packages.schedula</summary>
<pre>    rv = reductor(4)

tests/test_dispatcher.py: 97 warnings
tests/test_readme.py: 6 warnings
  /nix/store/h3778i8hp7kj1si0skvpdwb911qygn5w-python3.12-multiprocess-0.70.18/lib/python3.12/site-packages/multiprocess/popen_fork.py:66: DeprecationWarning: This process (pid=5438) is multi-threaded, use of fork() may lead to deadlocks in the child.
    self.pid = os.fork()

tests/utils/test_blue.py: 16 warnings
tests/utils/test_io.py: 4 warnings
  /nix/store/q54d9qqlb27g2nf9ja4zsa89gl7b3yvm-python3-3.12.11/lib/python3.12/pickle.py:576: DeprecationWarning: Pickle, copy, and deepcopy support will be removed from itertools in Python 3.14.
    rv = reduce(self.proto)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_readme.py::TestDoctest::runTest - AssertionError: 6 != 0 : (6, 55)
FAILED tests/utils/test_base.py::TestDoctest::runTest - AssertionError: 4 != 0 : (4, 25)
FAILED tests/utils/test_drw.py::TestDispatcherDraw::test_view - KeyError: 'plot'
FAILED tests/utils/test_web.py::TestDispatcherWeb::test_web - PermissionError: [Errno 1] Operation not permitted
FAILED tests/utils/test_web.py::TestDispatcherWeb::test_web_debug - PermissionError: [Errno 1] Operation not permitted
========== 5 failed, 95 passed, 1 deselected, 233 warnings in 39.46s ===========</pre>
</details>
<details>
<summary>python313Packages.schedula</summary>
<pre>    rv = reductor(4)

tests/test_dispatcher.py: 94 warnings
tests/test_readme.py: 6 warnings
  /nix/store/hrjf7irvbix29zhsllbqxw1d4lkwfld2-python3.13-multiprocess-0.70.18/lib/python3.13/site-packages/multiprocess/popen_fork.py:67: DeprecationWarning: This process (pid=5461) is multi-threaded, use of fork() may lead to deadlocks in the child.
    self.pid = os.fork()

tests/utils/test_blue.py: 16 warnings
tests/utils/test_io.py: 4 warnings
  /nix/store/j1spz18w5a10r9nj4wpkyh0cx730rs4z-python3-3.13.4/lib/python3.13/pickle.py:576: DeprecationWarning: Pickle, copy, and deepcopy support will be removed from itertools in Python 3.14.
    rv = reduce(self.proto)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
=========================== short test summary info ============================
FAILED tests/test_readme.py::TestDoctest::runTest - AssertionError: 6 != 0 : (6, 58)
FAILED tests/utils/test_base.py::TestDoctest::runTest - AssertionError: 4 != 0 : (4, 25)
FAILED tests/utils/test_drw.py::TestDispatcherDraw::test_view - KeyError: 'plot'
FAILED tests/utils/test_web.py::TestDispatcherWeb::test_web - PermissionError: [Errno 1] Operation not permitted
FAILED tests/utils/test_web.py::TestDispatcherWeb::test_web_debug - PermissionError: [Errno 1] Operation not permitted
========== 5 failed, 95 passed, 1 deselected, 230 warnings in 39.94s ===========</pre>
</details>
</details>

--- a/pkgs/development/python-modules/schedula/default.nix
+++ b/pkgs/development/python-modules/schedula/default.nix
@@ -1,0 +1,95 @@
+{
+  lib,
+  buildPythonPackage,
+  fetchFromGitHub,
+
+  # build-system
+  setuptools,
+
+  # optional-dependencies
+  dill,
+  flask,
+  graphviz,
+  multiprocess,
+  regex,
+  requests,
+  sphinx,
+  sphinx-click,
+
+  # tests
+  pytestCheckHook,
+  ddt,
+  cryptography,
+  schedula,
+}:
+
+buildPythonPackage rec {
+  pname = "schedula";
+  version = "1.5.62";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "vinci1it2000";
+    repo = "schedula";
+    tag = "v${version}";
+    hash = "sha256-erEUdiKV1MRwjVy3SKFneJVHp6gWEok7EWdv6v6HFGM=";
+  };
+
+  build-system = [ setuptools ];
+
+  optional-dependencies = rec {
+    # dev omitted, we have nativeCheckInputs for this
+    # form omitted, as it pulls in a kitchensink of deps, some not even packaged in nixpkgs
+    io = [ dill ];
+    parallel = [ multiprocess ];
+    plot = [
+      requests
+      graphviz
+      regex
+      flask
+    ];
+    sphinx = [
+      sphinx
+      sphinx-click
+    ] ++ plot;
+    web = [
+      requests
+      regex
+      flask
+    ];
+  };
+
+  nativeCheckInputs =
+    [
+      cryptography # doctests
+      ddt
+      sphinx
+      pytestCheckHook
+    ]
+    ++ schedula.optional-dependencies.io
+    ++ schedula.optional-dependencies.parallel
+    ++ schedula.optional-dependencies.plot;
+
+  disabledTests = [
+    # FAILED tests/test_setup.py::TestSetup::test_long_description - ModuleNotFoundError: No module named 'sphinxcontrib.writers'
+    "test_long_description"
+  ];
+
+  disabledTestPaths = [
+    # ERROR tests/utils/test_form.py::TestDispatcherForm::test_form1 - ModuleNotFoundError: No module named 'chromedriver_autoinstaller'
+    # ERROR tests/utils/test_form.py::TestDispatcherForm::test_form_stripe - ModuleNotFoundError: No module named 'chromedriver_autoinstaller'
+    "tests/utils/test_form.py"
+  ];
+
+  pythonImportsCheck = [ "schedula" ];
+
+  meta = {
+    description = "Smart function scheduler for dynamic flow-based programming";
+    homepage = "https://github.com/vinci1it2000/schedula";
+    changelog = "https://github.com/vinci1it2000/schedula/blob/v${version}/CHANGELOG.rst";
+    license = lib.licenses.eupl11;
+    maintainers = with lib.maintainers; [ flokli ];
+    # at least some tests fail on Darwin
+    platforms = lib.platforms.linux;
+  };
+}

--- a/pkgs/top-level/python-packages.nix
+++ b/pkgs/top-level/python-packages.nix
@@ -15851,6 +15851,8 @@ self: super: with self; {
 
   scenedetect = callPackage ../development/python-modules/scenedetect { };
 
+  schedula = callPackage ../development/python-modules/schedula { };
+
   schedule = callPackage ../development/python-modules/schedule { };
 
   scheduler = callPackage ../development/python-modules/scheduler { };

